lib/explorer/shared.ex

```diff
@@ -74,6 +74,10 @@ defmodule Explorer.Shared do
   def normalise_dtype(:u16), do: {:u, 16}
   def normalise_dtype(:u32), do: {:u, 32}
   def normalise_dtype(:u64), do: {:u, 64}
+
+  def normalise_dtype({:struct, map}) when is_map(map),
+    do: {:struct, map |> Map.to_list() |> List.keysort(0)}
```

Comment:
```suggestion
  def normalise_dtype({:struct, enum}),
    do: {:struct, enum |> Enum.map(fn {k, v} -> {to_string(k), normalize_dtype(v)} end) |> List.keysort(0)}
```

And then please check that `{:struct, a: :u32}` actually works. :) WDYT?

Reply:
Now it's working :)